build.zig

```diff
@@ -71,10 +71,22 @@ pub fn build(b: *Builder) !void {
     var ramdisk_files_al = ArrayList([]const u8).init(b.allocator);
     defer ramdisk_files_al.deinit();
 
-    // Add some test files for the ramdisk runtime tests
     if (test_mode == .Initialisation) {
+        // Add some test files for the ramdisk runtime tests
         try ramdisk_files_al.append("test/ramdisk_test1.txt");
         try ramdisk_files_al.append("test/ramdisk_test2.txt");
+    } else if (test_mode == .Scheduler) {
+        // Add some test files for the user mode runtime tests
+        const user_program = b.addAssemble("user_program", "test/user_program.s");
+        user_program.setOutputDir(b.cache_root);
+        user_program.setTarget(target);
+        user_program.setBuildMode(build_mode);
+        user_program.strip = true;
+
+        const copy_user_program = b.addSystemCommand(&[_][]const u8{ "objcopy", "-O", "binary", "zig-cache/user_program.o", "test/user_program" });
```

Comment:
I'm not sure about adding dependencies on a program like NASM. Most people are guaranteed to have objcopy installed, or not have any trouble installing it, since it's a GNU tool.

Reply:
I though objcopy would be a dependency anyway. But if not could make this this into a little helper function for compiling a flat binary asm file as I use nasm for the FAT heater (for the sub bootloader), so can use this instead.

Reply:
Yeah, I think objcopy is a better dependency to have, if we're choosing between objcopy and nasm, since most linux distributions and even WSL should have it installed by default, and those that don't shouldn't have any issue since it's a GNU tool. I can rebase on top of your fat32 builder patch and make a nice helper function :+1:

Reply:
Rebase? I assume this will be merged before my fat32 builder

Reply:
I doubt it, I'm having real trouble figuring out why the copyDataToVmm rt-test isn't working

Reply:
Also, I have seen you can use `addInstallRaw` like in #240

Reply:
Yep, that's much better :)

Reply:
I tried to sue `addInstallRaw` but an `.Obj` step (which is what's returned by `addAssemble()`) isn't supported atm. I'd like to use `addInstallRaw` once that is supported.

Reply:
No prob, what you have is good